pkg/utils/tsoutil/tso_dispatcher.go

```diff
@@ -153,14 +155,22 @@ func (s *TSODispatcher) dispatch(
 				if needUpdateServicePrimaryAddr && errs.IsLeaderChanged(err) {
 					tsoPrimaryWatchers[0].ForceLoad()
 				}
+				// Cancel the tsoQueue to prevent new requests from being accepted
 				tsoQueue.cancel(err)
```

Comment:
Just canceling the context does not prevent new requests from being processed completely. What about adding a field to indicated that this queue is closed so we can directly reject them from the very first begining? So we can just return error inside `DispatchRequest` before `tsoQueue.requestCh <- req` ASAP.